rxdiffadapter/src/main/java/com/revolut/rxdiffadapter/Preconditions.kt

```diff
@@ -0,0 +1,38 @@
+package com.revolut.rxdiffadapter
+
+import android.util.Log
+import androidx.recyclerview.widget.RecyclerView
+import com.revolut.recyclerkit.delegates.ListItem
+import com.revolut.recyclerkit.delegates.HasNestedDelegates
+import com.revolut.recyclerkit.delegates.RecyclerViewDelegate
+
+internal object Preconditions {
+
+    private const val TAG: String = "RX_DIFF_ADAPTER"
+
+    var throwErrorsEnabled = false
+
+    fun checkForDuplicateDelegates(delegates: List<RecyclerViewDelegate<out ListItem, out RecyclerView.ViewHolder>>) {
+        delegates.forEach { delegate ->
+            if (delegate is HasNestedDelegates) {
+                checkForDuplicateDelegates(delegate.delegates)
+            } else {
+                val possibleDuplicates = delegates.groupingBy { it::class.java }.eachCount()
```

Comment:
Don't you think grouping better be done by both class and viewType? (probably just viewType should be enough even), cause there might be a case where same Delegate class is used multiple times with different viewTypes

Reply:
Yes, quite rare case but possible I think if one delegate extend another, will update and also will log as error instead of warning,